includes/UserArray.php

```diff
@@ -21,13 +21,14 @@ static function newFromResult( $res ) {
 	 * @return UserArrayFromResult
 	 */
 	static function newFromIDs( $ids ) {
+	    global $wgExternalSharedDB;
 		$ids = array_map( 'intval', (array)$ids ); // paranoia
 		if ( !$ids ) {
 			// Database::select() doesn't like empty arrays
 			return new ArrayIterator(array());
 		}
-		$dbr = wfGetDB( DB_SLAVE );
-		$res = $dbr->select( 'user', '*', array( 'user_id' => $ids ),
+		$dbr = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+		$res = $dbr->select( '`user`', '*', array( 'user_id' => $ids ),
```

Comment:
Does `select` recognize arrays and change `WHERE user_id = 3` to `WHERE user_id IN (3, 4, 5)`?

Reply:
Yes, it does.

Reply:
Yes, it does.